controllers/machinedeployment_sync.go

```diff
@@ -159,7 +183,7 @@ func (r *MachineDeploymentReconciler) getNewMachineSet(ctx context.Context, d *c
 	newMS := clusterv1.MachineSet{
 		ObjectMeta: metav1.ObjectMeta{
 			// Make the name deterministic, to ensure idempotence
-			Name:            d.Name + "-" + apirand.SafeEncodeString(machineTemplateSpecHash),
+			Name:            d.Name + "-" + apirand.SafeEncodeString(machineDeploymentHash),
```

Comment:
Instead of modifying how the hash is calculated or any of the existing code, could we opt to move the name generation into a different function instead and optionally append the rolloutAfter fnv-hashed integer at the end of the name?

Reply:
I think this is pretty much describing what's implemented, I created a function see if that's what you're looking for.